val.py

```diff
@@ -245,10 +256,10 @@ def run(
     jdict, stats, ap, ap_class = [], [], [], []
     callbacks.run('on_val_start')
     pbar = tqdm(dataloader, desc=s, bar_format=TQDM_BAR_FORMAT)  # progress bar
-    for batch_i, (im, targets, paths, shapes) in enumerate(pbar):
+    for batch_i, (im0, im, targets, paths, shapes) in enumerate(pbar):
         callbacks.run('on_val_batch_start')
         if tagged_data:
-            confusion_matrix = TaggedConfusionMatrix(nc=nc)
+            confusion_matrix = TaggedConfusionMatrix(nc=nc) # TODO why in the for loop? Why not at line 247?
```

Comment:
because in the case of Tagged Validation, I want to recompute the matrix every iteration for every image.